### ISSUE TITLE
critcl 3.1.18 uninstall fails (Ubuntu)

### ISSUE THREAD

**Issue by apnadkarni** [andreas-kupries/critcl#104]
`tclsh build.tcl install` succeeds but a following `tclsh build.tcl uninstall` fails with error

```
couldn't open "/mnt/d/src/critcl-3.18-unix/lib/app-critcl/.. critcl critcl.tcl": no such file or directory
    while executing
"open $file r"
    (procedure "grep" line 2)
    invoked from within
"grep $file {*package provide*}"
    (procedure "version" line 2)
    invoked from within
"version [file join [file dirname $::me] lib $dir $vfile]"
    (procedure "_drop" line 32)
    invoked from within
"_drop {*}$args "
    (procedure "_uninstall" line 1)
    invoked from within
"_uninstall"
    ("eval" body line 1)
    invoked from within
"eval _$argv"
```

**Comment by apnadkarni** [comment]
This was with Tcl 8.6.8. I'm going to upgrade to 8.6.10 and retest. Tracing through the code, I do not understand how this could happen at all unless [file join] is broken.

**Comment by andreas-kupries** [comment]
Thank you. I will check that.

**Comment by andreas-kupries** [comment]
Confirmed all the issues. The only change to the fix needed is getting rid of `{*}`, because critcl still supports Tcl 8.4 in generated code, and its own code. Means that two other `{*}` in `build.tcl` have to go as well. ... And another two in `critcl-class/class.tcl`

**Comment by andreas-kupries** [comment]
Fixed with commit f178af9ca64538c7b9bc78cd6f973e662a9267db.
